### PR TITLE
Fix crash when no args passed in

### LINKED ISSUE #48: Ugly CLI exception if no command arguments are present
If I would just do the following command on the command line, with no arguments:

    create-react-native-module

it shows an ugly exception message like this:

```
/Users/brodybits/dev/nvs/node/8.16.0/x64/lib/node_modules/create-react-native-module/lib/normalized-options.js:18
    throw new Error('Please write your library\'s name');
    ^

Error: Please write your library's name
    at module.exports (/Users/brodybits/dev/nvs/node/8.16.0/x64/lib/node_modules/create-react-native-module/lib/normalized-options.js:18:11)
    at Object.func (/Users/brodybits/dev/nvs/node/8.16.0/x64/lib/node_modules/create-react-native-module/lib/cli-command.js:47:27)
    at Command.runAction (/Users/brodybits/dev/nvs/node/8.16.0/x64/lib/node_modules/create-react-native-module/cli.js:16:13)
    at Command.listener (/Users/brodybits/dev/nvs/node/8.16.0/x64/lib/node_modules/create-react-native-module/node_modules/commander/index.js:315:8)
    at emitNone (events.js:106:13)
    at Command.emit (events.js:208:7)
    at Command.parseArgs (/Users/brodybits/dev/nvs/node/8.16.0/x64/lib/node_modules/create-react-native-module/node_modules/commander/index.js:665:12)
    at Command.parse (/Users/brodybits/dev/nvs/node/8.16.0/x64/lib/node_modules/create-react-native-module/node_modules/commander/index.js:474:21)
    at Object.<anonymous> (/Users/brodybits/dev/nvs/node/8.16.0/x64/lib/node_modules/create-react-native-module/cli.js:27:9)
    at Module._compile (module.js:653:30)
```

I think it should show the usage help instead.

### PR DESCRIPTION
**Pull request by dlowder-salesforce**
Fix #48 .

Root cause: The `commander` module seems to try to detect this problem and push the `--help` argument on `process.argv`.  Unfortunately, that push is silently ignored, leading to the ugly error message issue.

Fix: Create our own args array, detect this condition, push `--help` onto the arg array, and then proceed with `program.parse()`.

Snapshot test also added.

### CODE REVIEW

File: lib/cli-program.js [hunk 1]
```diff
@@ -22,11 +22,8 @@ program
...
 
-program.parse(process.argv);
-
-// TBD non-functional error handling code
-// FUTURE TODO: handle missing argument error in the right place
-// (see issue #48)
-// if (!program.args.length) {
-//   program.help();
-// }
+const args = process.argv;
```
Comment (brodybits) [inline lib/cli-program.js]:
```suggestion
const args = [].concat(process.argv);
```

File: lib/cli-program.js [hunk 2]
```diff
@@ -22,11 +22,8 @@ program
...
+if (args.length === 2) {
```
Comment (brodybits) [inline lib/cli-program.js]:
```suggestion
if (args.length <= 2) {
```

Comment (brodybits) [inline lib/cli-program.js]: or better yet:

```suggestion
const MIN_ARGS_LENGTH = 3;

if (args.length < MIN_ARGS_LENGTH) {
```

(and maybe declare `const MIN_ARGS_LENGTH` right after the imports above)

**Comment by brodybits** [comment]
Now published, thanks again for this fix and for your review of #111. You can see that I raised some followup PRs, my apologies for the extra comments which I forgot to delete before submitting my review. And thanks for your contributions in general.

**Comment by dlowder-salesforce** [comment]
Very happy to help! This is a great tool and I intend to use it in my day job :) I don’t suppose you’d be interested in having this hosted in the react-native-community space?

**Pull request merged**